src/sentry/tagstore/snuba/backend.py

```diff
@@ -802,8 +804,26 @@ def get_tag_value_paginator_for_projects(
             versions = versions.order_by(*Release.SEMVER_COLS, "package").values_list(
                 "version", flat=True
             )[:1000]
+
+            seen = set()
+            formatted_versions = []
+            # We want to format versions here in a way that makes sense for autocomplete. So we
+            # - Only include package if we think the user entered a package
+            # - Exclude build number, since it's not used as part of filtering
+            # When we don't include package, this can result in duplicate version numbers, so we
+            # also de-dupe here. This can result in less than 1000 versions returned, but we
+            # typically use very few values so this works ok.
+            for version in versions:
+                formatted_version = version if include_package else version.split("@", 1)[1]
```

Comment:
It seems like there isn't a case where `include_package` is False and the version has a `@` in it, based on line 784. I assume I'm missing something?

Reply:
These are the versions returned from the database, might be better if I don't re-use the variable here to avoid confusion.